src/main/java/org/bookstore/controller/BookController.java

```diff
@@ -29,13 +33,24 @@ public List<BookDto> findAll() {
         return bookService.findAll();
     }
 
+    @ResponseStatus(HttpStatus.NO_CONTENT)
     @GetMapping("/{id}")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.getBookById(id);
     }
 
-    @GetMapping("/by-title")//by-name?title=bob
-    public List<BookDto> getAllByName(@RequestParam String title) {
-        return bookService.getAllByName(title);
+    @GetMapping("/by-title")//http://localhost:8080/api/books/by-title?title=matrix
```

Comment:
```suggestion
    @GetMapping
```
In RESTful design, the URL should ideally represent a resource and not include verbs or actions. RESTful URLs should be nouns and represent the resources being accessed. In this case, the URL is designed to find books by title, but it includes a query parameter that specifies the title.

A more RESTful approach would be to use a URL that represents the books resource and include the title as part of the resource path or as a query parameter in a more resource-centric way.